src/main/java/com/flippingutilities/model/CombinationFlip.java

```diff
@@ -0,0 +1,135 @@
+                                e.getValue().entrySet().stream().map(
+                                        entry -> Map.entry(entry.getKey(),
+                                                entry.getValue().clone())).
+                                        collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))))
+                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+        );
+    }
+
+    public long getProfit() {
+        return CombinationFlip.calculateProfit(parent, children);
+    }
+
+    private long getExpenseOrRevenue(boolean getRevenue) {
+        long totalValueOfChildren = children.values().stream().
+                flatMap(m -> m.values().stream()).
+                mapToLong(po -> po.offer.getPrice() * po.amountConsumed).
+                sum();
+        long totalValueOfParent = (long) parent.offer.getPrice() * parent.amountConsumed;
+        boolean parentIsBuy = parent.offer.isBuy();
+        if ((parentIsBuy && getRevenue) || (!parentIsBuy && !getRevenue)) {
+            return totalValueOfChildren;
+        }
+        return totalValueOfParent;
+    }
+
+    public long getExpense() {
+        return getExpenseOrRevenue(false);
+    }
+
+    public long getRevenue() {
+        return getExpenseOrRevenue(true);
+    }
+
+    public long getTaxPaid() {
+        return getOffers().stream().mapToLong(po -> po.getOffer().getTaxPaid()).sum();
+    }
+
+    public static long calculateProfit(PartialOffer parent, Map<Integer, Map<String, PartialOffer>> children) {
+        long totalValueOfChildren = children.values().stream().
+                flatMap(m -> m.values().stream()).
+                mapToLong(po -> po.offer.getPrice() * po.amountConsumed).
+                sum();
+        long totalValueOfParent = (long) parent.offer.getPrice() * parent.amountConsumed;
+        //if you buy the parent offer, you must be selling its children. Ex: buy a guthan set to deconstruct it and
+        //sell the individual pieces.
+        if (parent.offer.isBuy()) {
+            return totalValueOfChildren - totalValueOfParent;
+        }
+        //if you sell a parent, you must have constructed it from its children. Ex, buy the guthan set pieces to
+        //construct a set to sell it.
+        else {
+            return totalValueOfParent - totalValueOfChildren;
+        }
+    }
+
+    /**
+     * Gets all the partial offers contained in this combination flip.
+     */
+    public List<PartialOffer> getOffers() {
+        List<PartialOffer> offers = new ArrayList<>();
+        offers.add(parent);
+        offers.addAll(getChildrenOffers());
+        return offers;
+    }
+
+    /**
+     * Gets only the partial offers of  children
+     */
+    public List<PartialOffer> getChildrenOffers() {
+        return children.values().stream().
+                flatMap(
+                        m -> m.values().stream()).
+                collect(Collectors.toList());
```

Comment:
yea, I need to change it. There are a lot of styling issues, so what I'm gonna do is reformat the whole project under a consistent styling scheme and then push it to RL to be reviewed. I will do that in a later commit with JUST styling stuff so that it doesn't pollute feature reviews with tons of styling changes.